### DISCUSSION TITLE
Native mode deserialization failure for javax.ws.rs.core.Response$Status

### DISCUSSION THREAD

**Discussion by wabrit** [quarkusio/quarkus#25807] [Q&A] [answered]
I'm using Quarkus REST client to call an external service. which responds with a JSON payload:

```json
{
  "status", "NOT_FOUND",
  "statusCode", 404
}
```

I need to deserialize this into a Java class:

```java
import javax.ws.rs.core.Response;
...

@RegisterForReflection
public final class MyRestPayload {
  private Response.Status status;
  private int statusCode;

  public RestPayload() {}

  // plain getters and setters for status and statuscode
}
```

In non-native mode the following code, which transforms the jaxrs payload to my entity works fine:

```java
public Exception toThrowable(final Response response) {
    if (response.getStatus() == HttpStatus.SC_NOT_FOUND) {
      final MyRestPayload entity = response.readEntity(MyRestPayload.class);
      ...
    }
  }
}
```

But in native mode this fails because jackson seems to have lost its "smarts" about deserializing a string value to the `Response$Status` enum:

```
com.fasterxml.jackson.databind.exc.InvalidFormatException: 
  Cannot deserialize value of type `[javax.ws.rs](http://javax.ws.rs/).core.Response$Status` from number 404: 
  index value outside legal index range [0..39]\n at [Source: (ByteArrayInputStream); line: 1, column: 11] 
 (through reference chain: MyRestPayload[\"status\"])", ...
```

Any ideas how to fix this? I'm running Quarkus 2.6.2.Final

**Comment by wabrit** [comment, marked as answer]
OK ignore me - it turned out to have nothing to do with native code. Sorry!

**Reply by pjsilva613** [reply]
hola, buen dia, actualmente presento el mismo error, lo lograste solucionar?, como lo solucionaste?
